### PR TITLE
Move settings menu to options menu

### PR DESCRIPTION
**Pull request by alexfriesen**
I removed the settings button from the bottom menu and added it to the options menu. 
Also the options menu creates the buttons via code.
pp_menu_options

### CODE REVIEW

**Comment by sboigelot** [comment]
Is your new UI not using anymore the FurnitureBuildMenu? (this is already merged but probably related to your previous change in the UI)

While saving the settings, we have an exception:

`NullReferenceException: Object reference not set to an instance of an object
DialogBoxSettings.OnClickSave () (at Assets/Scripts/UI/DialogBox/Settings/DialogBoxSettings.cs:87)
DialogBoxSettings.<OnEnable>m__51 () (at Assets/Scripts/UI/DialogBox/Settings/DialogBoxSettings.cs:120)
UnityEngine.Events.InvokableCall.Invoke (System.Object[] args) (at C:/buildslave/unity/build/Runtime/Export/UnityEvent.cs:153)
UnityEngine.Events.InvokableCallList.Invoke (System.Object[] parameters) (at C:/buildslave/unity/build/Runtime/Export/UnityEvent.cs:630)
UnityEngine.Events.UnityEventBase.Invoke (System.Object[] parameters) (at C:/buildslave/unity/build/Runtime/Export/UnityEvent.cs:765)
UnityEngine.Events.UnityEvent.Invoke () (at C:/buildslave/unity/build/Runtime/Export/UnityEvent_0.cs:53)
UnityEngine.UI.Button.Press () (at C:/buildslave/unity/build/Extensions/guisystem/UnityEngine.UI/UI/Core/Button.cs:35)
UnityEngine.UI.Button.OnPointerClick (UnityEngine.EventSystems.PointerEventData eventData) (at C:/buildslave/unity/build/Extensions/guisystem/UnityEngine.UI/UI/Core/Button.cs:44)
UnityEngine.EventSystems.ExecuteEvents.Execute (IPointerClickHandler handler, UnityEngine.EventSystems.BaseEventData eventData) (at C:/buildslave/unity/build/Extensions/guisystem/UnityEngine.UI/EventSystem/ExecuteEvents.cs:52)
UnityEngine.EventSystems.ExecuteEvents.Execute[IPointerClickHandler] (UnityEngine.GameObject target, UnityEngine.EventSystems.BaseEventData eventData, UnityEngine.EventSystems.EventFunction`1 functor) (at C:/buildslave/unity/build/Extensions/guisystem/UnityEngine.UI/EventSystem/ExecuteEvents.cs:269)
UnityEngine.EventSystems.EventSystem:Update()
`

This is because the instance in
`FurnitureBuildMenu.instance.RebuildMenuButtons(developerModeToggle.isOn);`
is null, FurnitureBuildMenu is never used.

**Comment by sboigelot** [comment]
If that line of code is not required anymore, just get rid of it. But it may be, I don't know how your build menu refresh. I suppose it is because some building can't be build if you are not in dev mode.

**Comment by alexfriesen** [comment]
fixed. 
I need to remove some old UI stuff in the next PR, but I don't know if that is going to breaking others PR's...

File: Assets/Scripts/UI/DialogBox/Options/DialogBoxOptions.cs [hunk 1]
```diff
@@ -31,45 +27,83 @@ public void OnButtonLoadGame()
...
+        });
+
+        GameObject quitButton = CreateButtonGO(buttonPrefab, "Quit", "menu_quit");
+        quitButton.GetComponent<Button>().onClick.AddListener(delegate
+        {
+            OnButtonQuitGame();
+        });
+    }
+
+    private GameObject CreateButtonGO(Object buttonPrefab, string name, string localizationCode)
```
Comment (TomMalbran) [inline Assets/Scripts/UI/DialogBox/Options/DialogBoxOptions.cs]: is the name param required? We could just use the localizationCode.

Comment (sboigelot) [inline Assets/Scripts/UI/DialogBox/Options/DialogBoxOptions.cs]: I think this is out of the scope of this PR

**Comment by sboigelot** [comment]
You have a lot of stylecop errors

`Warning	28	SA1515 : CSharp.Layout : A single-line comment must be preceded by a blank line or another single-line comment, or must be the first item in its scope. To ignore this error when commenting out a line of code, begin the comment with '////' rather than '//'.	D:\git\ProjectPorcupine\Assets\Scripts\Controllers\WorldController.cs	226	1	ProjectPorcupine.CSharp
Warning	29	SA1005 : CSharp.Spacing : The comment must start with a single space. To ignore this error when commenting out a line of code, begin the comment with '////' rather than '//'.	D:\git\ProjectPorcupine\Assets\Scripts\Controllers\WorldController.cs	226	10	ProjectPorcupine.CSharp
Warning	30	SA1121 : CSharp.Readability : Use the built-in type alias 'object' rather than Object or System.Object.	D:\git\ProjectPorcupine\Assets\Scripts\UI\DialogBox\Options\DialogBoxOptions.cs	50	1	ProjectPorcupine.CSharp
Warning	31	SA1121 : CSharp.Readability : Use the built-in type alias 'object' rather than Object or System.Object.	D:\git\ProjectPorcupine\Assets\Scripts\UI\DialogBox\Options\DialogBoxOptions.cs	89	1	ProjectPorcupine.CSharp
Warning	32	SA1515 : CSharp.Layout : A single-line comment must be preceded by a blank line or another single-line comment, or must be the first item in its scope. To ignore this error when commenting out a line of code, begin the comment with '////' rather than '//'.	D:\git\ProjectPorcupine\Assets\Scripts\UI\DialogBox\Settings\DialogBoxSettings.cs	87	1	ProjectPorcupine.CSharp
Warning	33	SA1005 : CSharp.Spacing : The comment must start with a single space. To ignore this error when commenting out a line of code, begin the comment with '////' rather than '//'.	D:\git\ProjectPorcupine\Assets\Scripts\UI\DialogBox\Settings\DialogBoxSettings.cs	87	10	ProjectPorcupine.CSharp
Warning	34	SA1121 : CSharp.Readability : Use the built-in type alias 'object' rather than Object or System.Object.	D:\git\ProjectPorcupine\Assets\Scripts\UI\InGameUI\MenuLeft\ConstructionMenu.cs	68	1	ProjectPorcupine.CSharp
Warning	35	SA1121 : CSharp.Readability : Use the built-in type alias 'object' rather than Object or System.Object.	D:\git\ProjectPorcupine\Assets\Scripts\UI\InGameUI\MenuLeft\ConstructionMenu.cs	114	1	ProjectPorcupine.CSharp
Warning	36	SA1121 : CSharp.Readability : Use the built-in type alias 'object' rather than Object or System.Object.	D:\git\ProjectPorcupine\Assets\Scripts\UI\InGameUI\MenuLeft\ConstructionMenu.cs	151	1	ProjectPorcupine.CSharp
`

Can you check them?

Also, you commented out some code, just delete it instead. We shouldn't keep commented code.

File: Assets/Scripts/UI/DialogBox/Options/DialogBoxOptions.cs (hunk 1 above)
Comment (alexfriesen) [inline Assets/Scripts/UI/DialogBox/Options/DialogBoxOptions.cs]: The name will set a understandable tag in the hierarchy tree, localization code would look inconsistent.

**Comment by alexfriesen** [comment]
One of the StyleCop errors is for this line:
Object buttonPrefab = Resources.Load("UI/Components/MenuButton");

Resources.Load generates an Object but stylecop recomments to change it to a object. But this doesnt work with unity. 

What should I do with this case? Can I ignore and suppress this error or is there a way to fix it properly?

**Comment by Grenkin1988** [comment]
@alexfriesen Could it be used as:
`GameObject instance = Resources.Load("UI/Components/MenuButton", typeof(GameObject)) as GameObject;` ?

**Comment by TomMalbran** [comment]
What about:
```UnityEngine.Object buttonPrefab = Resources.Load("UI/Components/MenuButton");```

It is how it was fixed here: https://github.com/TeamPorcupine/ProjectPorcupine/pull/1280/files#diff-f0b4785c41e42829671ac4f40b9a0785R155

**Comment by NogginBops** [comment]
It should be `UnityEngine.Object` it's confusing to with `System.Object` which should be written as `object`

**Comment by sboigelot** [comment]
This is just a stylecop rule confusing` UnityEngine.Object` with `System.Object` and asking to use `object`, I think using `UnityEngine.Object` is fine in this context.

**Comment by koosemose** [comment]
Conflicted.

**Comment by alexfriesen** [comment]
all fixed

**Pull request merged**